### ISSUE TITLE
[BUG] ERROR :: root :: sync.py :: 116 :: Password is not stored in keyring. Please save the password in keyring.

### ISSUE THREAD

**Issue by knmax** [mandarons/icloud-docker#256]
**Describe the bug**
ERROR :: root :: sync.py :: 116 :: Password is not stored in keyring. Please save the password in keyring.

**To Reproduce**

1. login with 2FA
2. See error

**Expected behavior**
successful login

**Screenshots**
<img width="1014" alt="image" src="https://github.com/user-attachments/assets/d85ef482-aa11-43ad-9a19-c6fb3e7ac578">

**Configuration**
default configuration as I was upgrading from an older release and nothing worked

**Comment by mandarons** [comment]
Please try:

docker exec -it --user=abc icloud /bin/sh -c "icloud --username=<icloud-username> --session-directory=/config/session_data"

**Comment by knmax** [comment]
> Please try:
> 
> docker exec -it --user=abc icloud /bin/sh -c "icloud --username= --session-directory=/config/session_data"

I already did that to login with 2FA - otherwise no login is possible with 2FA or am I missing anything?

**Comment by mandarons** [comment]
Did you run the command with '--user=abc' flag?

**Comment by knmax** [comment]
yes

**Comment by mandarons** [comment]
Looks like your user id is 0 and group id is 911.

Try with '--user=0:911' flag.

**Comment by knmax** [comment]
just tried, but still the same message.
Strange thing is, output of id -u $user and id -g $user is both 0 - I filled that in the docker-compose.yaml but it get's ignored when starting the container

**Comment by mandarons** [comment]
May be try removing PUID and PGID from your docker configuration?

**Comment by knmax** [comment]
When trying to start the container without PUID and GUID I get this:
<img width="707" alt="image" src="https://github.com/user-attachments/assets/9886794c-951a-4e6b-8277-74256bcfa0ca">
I found that I have a spelling error in my docker-compose.yaml I wrote PUID and GUID instead of PUID and PGID
After correcting this I finally have the correct PUID and PGID shown when running docker compose up, but I'm still getting the message that the password is not stored in keyring.
I found a different issue which mentions not using the ENV_ICLOUD_PASSWORD - I'm also not using it as I don't want my iCloud Password in clear text in the configfiles, might this be some kind of issue?

**Comment by knmax** [comment]
I found the issue - as I was already using a older version in the past, I did not know that the config-path must be mentioned in the .env.icloud file, therefore it was not in there.
I was now able to start the container and get a successful authentication even without having the ENV_ICLOUD_PASSWORD variable filled, now I am being served with the following error message.
<img width="1312" alt="image" src="https://github.com/user-attachments/assets/ae4e8121-2c6f-41cc-97f9-d9da9998bc9b">

Also the part where it "syncs" my drive, I have over 20 folders in my Drive and I don't believe that it syncs all those folders in a matter of a few seconds.

**Comment by mandarons** [comment]
The error that you're seeing is indicating that default (app provided) `config.yaml` file is being used. Default [config.yaml](https://github.com/mandarons/icloud-docker/blob/main/config.yaml) has `album 1` in it as an example filter. 

Please make sure you're mounting local `config/` to `/config` folder in the container (i.e. `${PWD}/config:/config # Must contain config.yaml`), as mentioned in the [README](https://github.com/mandarons/icloud-docker/blob/main/README.md).

Going through [README](https://github.com/mandarons/icloud-docker/blob/main/README.md) thoroughly just to double-check if everything has been set up correctly would be good idea.

**Comment by knmax** [comment]
The problem was that I was in the wrong folder when starting the container and therefore there was truly no config.yaml present...
I managed to get past the other Error but, now it seems I'm presented with this Error:
<img width="969" alt="image" src="https://github.com/user-attachments/assets/bae25bc6-2f1e-45cd-b003-6a29fd135faf">

**Comment by mandarons** [comment]
This error is about your config file. There seems to be misalignment at line 39 and 47.

**Comment by knmax** [comment]
I just directly copied the example config file into my setup and just removed the lines that are not needed for my setup, please have a look.
Now I get the same message about line 40 and 44
<img width="504" alt="image" src="https://github.com/user-attachments/assets/7095360c-5427-4cf5-9e54-7e92d2996e68">
```dockerfile
app:
  logger:
    # level - debug, info (default), warning or error
    level: "info"
    # log filename icloud.log (default)
    filename: "/config/icloud.log"
  credentials:
    # iCloud drive username
    username: " myappleid"
    # Retry login interval - default is 10 minutes, specifying -1 will retry login only once and exit
    retry_login_interval: 600
  # Drive destination
  root: "/icloud"
  discord:
  # webhook_url: <your server webhook URL here>
  # username: icloud-docker #or any other name you prefer
  telegram:
  # bot_token: <your Telegram bot token>
  # chat_id: <your Telegram user or chat ID>
  smtp:
    ## If you want to receive email notifications about expired/missing 2FA credentials then uncomment
    # email: "user@test.com"
    ## optional, to email address. Default is sender email.
    # to: "receiver@test.com"
    # password:
    # host: "smtp.test.com"
    # port: 587
    # If your email provider doesn't handle TLS
    # no_tls: true
  region: global # For China server users, set this to - china (default: global)
drive:
  destination: "drive"
  remove_obsolete: false
  sync_interval: 300
  ignore:
    # When specifying folder paths, append it with /*
    - "node_modules/*"
    - "*.md"
photos:
  destination: "photos"
  remove_obsolete: false
  sync_interval: 500
  all_albums: true # Optional, default false. If true preserve album structure. If same photo is in multiple albums creates duplicates on filesystem
    file_sizes: # valid values are original, medium and/or thumb
      - "original"
      # - "medium"
      # - "thumb"
``` 


>

**Comment by mandarons** [comment]
You're missing 'filters:' just above 'file_sizes:' in your config file.

**Comment by knmax** [comment]
> This error is about your config file. There seems to be misalignment at line 39 and 47.

Ok, I thought I could also remove that if I just want to download all, will retry in the evening.

**Comment by knmax** [comment]
Thanks for the point in the right direction.
The sync is now working
One thing which would be good to add in the instructions is how to set a different docker volume for storage as I guess some people would want to store the drive and pictures in a different volume.

**Comment by mandarons** [comment]
Mounting `/icloud/drive` and `/icloud/photos` to different paths in docker configuration should do it.
